templates/matched_plugin_id.j2

```diff
@@ -14,7 +14,7 @@
 {% set consumer_id = consumer.json.id %}
 {% endif %}
 {# Compare values #}
-{% if name == p.name and service_id == p_service_id and consumer_id == p_consumer_id %}
+{% if ( (name == p.name and service_id == p_service_id) or (name == p.name and consumer_id == p_consumer_id) ) %}
```

Comment:
The patch won't work unfortunately. 
The current logic is valid. It takes care of a case where `customer_id` and/or `service_id` is not specified in the config. As you'll observe in the comment above a plugin object config is identified by `plugin_name + service_id + consumer_id`

```
{# The purpose of this template is to find a matching plugin object based on: #}
{#    * plugin_name + service_id + consumer_id                                #}
```
Furthermore, the default values of `service_id` and `consumer_id` are set to empty string, they get overridden appropriately and the new values used to compare against `p_service_id` and `p_consumer_id` respectively.

Reply:
Hey @wunzeco , 
yes you're right i think the issue of updating plugings (ref https://github.com/wunzeco/ansible-kong/issues/16 ) related to kong pagination
```
- name: Get list of plugin objects configured in kong
  uri:
    url: "{{ kong_admin_api_url|default('http://localhost:8001') }}/plugins"
  register: plugins
```
the variable `plugins` here contain only first 100 result by default